source/ota.c

```diff
@@ -1772,62 +1580,62 @@ static OtaJobParseErr_t parseJobDocFromCustomCallback( const char * pJson,
 {
     OtaErr_t otaErr = OTA_ERR_NONE;
 
-    /* We have an unknown job parser error. Check to see if we can pass control to a callback for parsing */
-    OtaJobParseErr_t err = otaAgent.palCallbacks.customJobCallback( pJson, messageLength );
+    OtaJobParseErr_t err = OtaJobParseErrNone;
 
-    if( err == OtaJobParseErrNone )
+    /* We have an unknown job parser error. Check to see if we can pass control to a callback for parsing */
+    if( otaAgent.customJobCallback != NULL )
     {
-        /* Custom job was parsed by external callback successfully. Grab the job name from the file
-         *  context and save that in the ota agent */
-        if( pFileContext->pJobName != NULL )
-        {
-            otaAgent.pOtaSingletonActiveJobName = pFileContext->pJobName;
-            pFileContext->pJobName = NULL;
-            otaErr = otaControlInterface.updateJobStatus( &otaAgent,
-                                                          JobStatusSucceeded,
-                                                          JobReasonAccepted,
-                                                          0 );
+        err = otaAgent.customJobCallback( pJson, messageLength );
 
-            if( otaErr != OTA_ERR_NONE )
+        if( err == OtaJobParseErrNone )
+        {
+            /* Custom job was parsed by external callback successfully. Grab the job name from the file
+             *  context and save that in the ota agent */
+            if( strlen( pFileContext->pJobName ) > 0 )
             {
-                LogError( ( "Failed to update job status: "
-                            "updateJobStatus returned error: "
-                            "OtaErr_t=%d",
-                            otaErr ) );
-            }
+                memcpy( otaAgent.pActiveJobName, otaAgent.fileContext.pJobName, OTA_JOB_ID_MAX_SIZE );
+                otaErr = otaControlInterface.updateJobStatus( &otaAgent,
+                                                              JobStatusSucceeded,
+                                                              JobReasonAccepted,
+                                                              0 );
 
-            /* Everything looks OK. Set final context structure to start OTA. */
-            **pFinalFile = *pFileContext;
-            LogInfo( ( "Job document parsed from external callback" ) );
+                /* Everything looks OK. Set final context structure to start OTA. */
+                **pFinalFile = *pFileContext;
+                LogInfo( ( "Job document parsed from external callback" ) );
 
-            /* We don't need the job name memory anymore since we're done with this job. */
-            free( otaAgent.pOtaSingletonActiveJobName );
-            otaAgent.pOtaSingletonActiveJobName = NULL;
+                /* We don't need the job name memory anymore since we're done with this job. */
+                memset( otaAgent.pActiveJobName, 0, OTA_JOB_ID_MAX_SIZE );
```

Comment:
I see `otaAgent.pActiveJobName` used in a lot of functions.  How did we go from populating it 10 lines earlier to being done with it here?

Reply:
Here the job status is updated with success( based on the custom job parsing ) and thats why we dont need the context anymore.